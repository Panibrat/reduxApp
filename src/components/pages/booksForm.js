import React from 'react';
import {Well, Panel, FormControl, FormGroup, ControlLabel, Button} from 'react-bootstrap';


import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {findDOMNode} from 'react-dom';
import {postBooks, deleteBooks} from '../../actions/booksActions';


export class BooksForm extends React.Component {
  constructor(props) {
      super(props);
      this.handleSubmit = this.handleSubmit.bind(this);
      this.onDelete = this.onDelete.bind(this);
    }

  handleSubmit(){
    const book = [
      {
        //id: 12121,
        title: findDOMNode(this.refs.title).value,
        //title: this.refs.title.value + '!?!?', - doesn't work with react-bootstrap component
        description:  findDOMNode(this.refs.description).value,
        price: findDOMNode(this.refs.price).value
      }
    ];
    this.props.postBooks(book);
  }

    onDelete(){
      var id = findDOMNode(this.refs.delete).value
      var bookToDelete = {_id: id}
      this.props.deleteBooks(bookToDelete);
  }

  render() {
    const booksList = this.props.books.map((bookItem)=> {
      return   <option key={bookItem._id} > {bookItem._id}</option>
    })
    return (
      <Well>
        <Panel>
          <FormGroup controlId="title">
            <ControlLabel>Title</ControlLabel>
            <FormControl
              type="text"
              ref="title"
              placeholder="Enter title"
          />
          </FormGroup>
          <FormGroup controlId="description">
            <ControlLabel>Description</ControlLabel>
            <FormControl
              type="text"
              ref="description"
              placeholder="Enter description"
          />
          </FormGroup>
          <FormGroup controlId="price">
            <ControlLabel>Price</ControlLabel>
            <FormControl
              type="text"
              ref="price"
              placeholder="Enter price"
          />
          </FormGroup>
          <Button onClick={this.handleSubmit} bsStyle="primary">SAVE</Button>
        </Panel>
        <Panel>
          <FormGroup controlId="formControlsSelect">
            <ControlLabel>Select book to deleteeee</ControlLabel>
            <FormControl ref="delete" componentClass="select" placeholder="select">
              <option value="select">select</option>
              {booksList}
            </FormControl>
          </FormGroup>
          <Button onClick={this.onDelete} bsStyle="danger">
            Delete book
          </Button>
        </Panel>
      </Well>
    )
  }
}

function mapStateToProps(state) {
  return {
    books: state.books.books
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    postBooks: postBooks,
    deleteBooks: deleteBooks
  }, dispatch)
};

export default connect(mapStateToProps, mapDispatchToProps)(BooksForm);
