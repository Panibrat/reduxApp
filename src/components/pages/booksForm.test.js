import React from 'react';
import {shallow} from 'enzyme';
import {BooksForm} from './booksForm';

const book = [
  {
    _id: 1,
    title: 'First Book',
    description: 'this is the book description ',
    price: 10.25,
    quantity: 5
  }
]
test('should render booksForm correctly ', () => {
  const wrapper = shallow(<BooksForm />);
  expect(wrapper).toMatchSnapshot();

});

test('should simulate submit button from booksForm', () => {
  const wrapper = shallow(<BooksForm />);
  wrapper.find('Button').simulate('submit');
  expect(wrapper).toMatchSnapshot();

});
