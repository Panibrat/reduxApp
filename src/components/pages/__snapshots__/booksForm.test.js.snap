// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render booksForm correctly  1`] = `
<Well
  bsClass="well"
>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}
  >
    <FormGroup
      bsClass="form-group"
      controlId="title"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Title
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        placeholder="Enter title"
        type="text"
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      controlId="description"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Description
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        placeholder="Enter description"
        type="text"
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      controlId="price"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Price
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        placeholder="Enter price"
        type="text"
      />
    </FormGroup>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      SAVE
    </Button>
  </Panel>
</Well>
`;

exports[`should simulate submit button from booksForm 1`] = `
<Well
  bsClass="well"
>
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}
  >
    <FormGroup
      bsClass="form-group"
      controlId="title"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Title
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        placeholder="Enter title"
        type="text"
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      controlId="description"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Description
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        placeholder="Enter description"
        type="text"
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      controlId="price"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Price
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        placeholder="Enter price"
        type="text"
      />
    </FormGroup>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      SAVE
    </Button>
  </Panel>
</Well>
`;
